sympy/core/add.py

```diff
@@ -576,6 +580,8 @@ def _eval_is_positive(self):
             v = _monotonic_sign(a)
             if v is not None:
                 s = v + c
+                if s == self:
+                    return None
                 if s.is_positive and a.is_nonnegative:
```

Comment:
It might be expedient to add a test before the `s == self` line. This will be the case when `v == a` is an `_eps` created by `_monotonic_sign`. Its sign can be tested, and if the constant part `c` happens to be non-negative, the sign of `s = v + c` can be determined:
```
    if v.is_positive and c.is_nonnegative:
        return True
```